analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PubliclyWritableDirectories.cs

```diff
@@ -0,0 +1,100 @@
+            tmp = Environment.GetEnvironmentVariable("TEMP"); // Noncompliant
+            tmp = Environment.ExpandEnvironmentVariables("%TMPDIR%"); // Noncompliant
+            tmp = Environment.ExpandEnvironmentVariables("%TMP%"); // Noncompliant
+            tmp = Environment.ExpandEnvironmentVariables("%TEMP%"); // Noncompliant
+            tmp = "%USERPROFILE%\\AppData\\Local\\Temp\\f"; // Noncompliant
+            tmp = "%TEMP%\\f"; // Noncompliant
+            tmp = "%TMP%\\f"; // Noncompliant
+            tmp = "%TMPDIR%\\f"; // Noncompliant
+//                ^^^^^^^^^^^^^
+
+            // Common
+            using (var tmpDir = new StreamReader("/tmp/f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/tmp")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/var/tmp/f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/usr/tmp/f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/dev/shm/f")) { }; // Noncompliant
+
+            // Linux
+            using (var tmpDir = new StreamReader("/dev/mqueue/f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/run/lock/f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/var/run/lock/f")) { }; // Noncompliant
+
+            // MacOS
+            using (var tmpDir = new StreamReader("/Library/Caches/f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/Users/Shared/f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/private/tmp/f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("/private/var/tmp/f")) { }; // Noncompliant
+
+            // Windows
+            using (var tmpDir = new StreamReader("C:\\Windows\\Temp\\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("C:\\Temp\\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("C:\\TEMP\\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("C:\\TMP\\f")) { }; // Noncompliant
+
+            // Variates
+            using (var tmpDir = new StreamReader(@"/tmp/f")) { };  // Noncompliant
+            using (var tmpDir = new StreamReader("D:\\Windows\\Temp\\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("\\\\Windows\\Temp\\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader("\\Windows\\Temp\\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader(@"C:\Windows\Temp\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader(@"D:\Windows\Temp\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader(@"\\Windows\Temp\f")) { }; // Noncompliant
+            using (var tmpDir = new StreamReader(@"\Windows\Temp\f")) { }; // Noncompliant
+            tmp = "/tmp/f"; // Noncompliant
+            tmp = "/tEmP/f"; // Noncompliant
```

Comment:
Why is this specific case noncompliant? Linux file system is case sensitive so `/tEmP/` is not `/temp/`

Reply:
From what I understood, all the cases after the //variates comment are not linux only so they should have a case insensitive check.